samples/socket-api/wasm-src/timeout_client.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2022 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+
+int
+main(int argc, char *argv[])
+{
+    int socket_fd;
+    struct sockaddr_in addr;
+    struct timeval tv = { 0, 1 };
+    const int snd_buf_len = 8;
+    const int data_buf_len = 1000000;
+    char *buffer = (char *)malloc(sizeof(char) * data_buf_len);
+    int result;
+    socklen_t opt_len = sizeof(snd_buf_len);
+    struct timeval snd_start_time, snd_end_time;
+    int bool_opt = 1;
+
+    if (buffer == NULL) {
+        perror("Allocation failed, please re-run with larger heap size");
+        return EXIT_FAILURE;
+    }
+
+    /* 127.0.0.1:1234 */
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(1234);
+    addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+
+    if ((socket_fd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        perror("Create socket failed");
+        free(buffer);
+        return EXIT_FAILURE;
```

Comment:
Same as above, had better define `fail1`, `fail2` labels after L112:
```C
fail2:
    close(socket_fd);
fail1:
    free(buffer);
    return EXIT_FAILURE;
```
And here `goto fail1`, L51/L58/L65/L74/... `goto fail2;`